packages/react/src/useFetch.ts

```diff
@@ -117,27 +145,73 @@ export function useFetch<T = string>(path: string, options?: FetchHookOptions):
       try {
         const { json: _json, stream: _stream, ...fetchOptions } = mergedOptions;
         // make the fetch call using GadgetConnection to pass along auth and other headers
-        response = await connection.fetch(path, fetchOptions);
+        response = await connection.fetch(path, { signal: abortController.signal, ...fetchOptions });
         if (!response.ok) {
           throw new Error(response.statusText);
         }
 
+        let dispatchData = true;
+
         if (mergedOptions.json) {
           data = await response.json();
+        } else if (typeof mergedOptions.stream === "string") {
+          dispatchData = false;
+          const decodedStream = response.body!.pipeThrough(
+            new TextDecoderStream(mergedOptions.stream === "string" ? "utf8" : mergedOptions.stream)
+          );
+
+          const [responseStream, updateStream] = decodedStream.tee();
+
+          data = responseStream;
+          const decodedStreamReader = updateStream.getReader();
+
+          decodedStreamReader.closed.catch((error) => {
+            if (sendId === sendCounter.current) {
+              dispatchError(mounted, dispatch, error, response);
+            }
+          });
+
+          dispatch({ type: "fetched", payload: "" as any });
+
+          (async () => {
+            let responseText = "";
+            let done = false;
+
+            while (!done) {
+              if (!(mounted.current && sendId === sendCounter.current)) {
+                abortController.abort("Request cancelled");
+                await decodedStreamReader.cancel();
+                break;
+              }
+
+              const { value, done: _done } = await decodedStreamReader.read();
+              done = _done;
+
+              if (value) {
+                responseText += value;
+              }
+
+              if (mounted.current && sendId === sendCounter.current) {
+                dispatch({ type: "update", payload: responseText as any });
+              }
```

Comment:
nit: if `!value`, we still dispatch. Probably move inside the `if (value)`?

Alternatively, the "update" dispatch could take `value` and only return a new state object if `!!value`.